### DISCUSSION TITLE
Build environmental issues & run WAR error

### DISCUSSION THREAD

**Discussion by NicoChiGu** [alfio-event/alf.io#982] [General]
Hi!
       Thank you very much! I can use such a great ticket system.
       .WAR run smoothly on heroku.
       
But i encounter problems when running local machine.

> System OS: Ubuntu Server 18.0.4.5 LTS X64
> Existing environment: OPENJDK-11 
> SQL: postgresql 11.0

Hundred errors in normal build
       How to solve these problems. [LOGFILE](http://ys-i.ys168.com/616727551/jsgOVTk52556J267UMK/alfio.log)
       If run ' java -jar alfio-2.0-M3.1-boot.war ' 
         
> 
> ```
> - 21:52:20.468 [main] INFO  alfio.config.SpringBootLauncher - Starting SpringBootLauncher on ubuntuServer with PID 26307 (/root/alf.io-2.0-M3-maintenance/alfio-2.0-M3.1-boot.war started by root in /root/alf.io-2.0-M3-maintenance)
> - 21:52:20.493 [main] INFO  alfio.config.SpringBootLauncher - The following profiles are active: spring-boot
> - 21:52:25.046 [main] WARN org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Jetty web server
> -  21:52:25.068 [main] ERROR org.springframework.boot.SpringApplication - Application run failed
>  (Omit) 
> ```

very thankful!

**Comment by cbellone** [comment]
Hi! 
Thank you for using alf.io

You should add an `application.properties` file, where alf.io can find the URL of your database and the credentials to access it.
You can find a sample `application.properties` here: https://github.com/alfio-event/alf.io/issues/51#issuecomment-98058075

Thanks!

Edit: I've noticed the logfile link which confirmed that you need to add the application.properties file in your folder

**Reply by NicoChiGu** [reply]
Thank you very much! Already running.

Because the former problem is closed in github.I asked this question after looking up the information for a few days without results.

**Comment by NicoChiGu** [comment]
If WEB run on the heroku
Is it possible to share privately constructed SQL. 
For example, **heroku(web)----->My dbserver(SQL)<-----My web server**

**Reply by cbellone** [reply]
I am not sure you can do that, at least with a basic plan, but it's worth asking heroku's support if it's possible.

**Reply by NicoChiGu** [reply]
My idea is that it works.
By directly modifying ‘Config Vars’

Thank you for answering my question!
